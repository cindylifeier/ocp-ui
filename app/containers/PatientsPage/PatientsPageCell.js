import { Cell } from 'styled-css-grid';

const PatientsPageCell = Cell.extend`
  background-color: #fff;
  color: rgb(51, 51, 51);
  border-radius: 5px;
  padding: 3px;
  font-size: 100%;
`;

PatientsPageCell.propTypes = {};

export default PatientsPageCell;
