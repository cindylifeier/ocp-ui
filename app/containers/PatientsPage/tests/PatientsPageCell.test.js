import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import PatientsPageCell from '../PatientsPageCell';

configure({ adapter: new Adapter() });

describe('<PatientsPageCell />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<PatientsPageCell>{children}</PatientsPageCell>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<PatientsPageCell>{children}</PatientsPageCell>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('style tests', () => {
    it('should have styles', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<PatientsPageCell>{children}</PatientsPageCell>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('background-color', '#fff');
      expect(renderedComponent).toHaveStyleRule('color', 'rgb(51, 51, 51)');
      expect(renderedComponent).toHaveStyleRule('border-radius', '5px');
      expect(renderedComponent).toHaveStyleRule('padding', '3px');
      expect(renderedComponent).toHaveStyleRule('font-size', '100%');
    });
  });
});
