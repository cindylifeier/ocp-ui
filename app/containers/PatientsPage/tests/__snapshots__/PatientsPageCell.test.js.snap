// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PatientsPageCell /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.section>
    <span>
      test
    </span>
  </styled.section>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        test
      </span>,
      "className": "sc-bdVaJa ciCvCZ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "span",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          test
        </span>,
        "className": "sc-bdVaJa ciCvCZ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
